includes/SVG.php

```diff
@@ -0,0 +1,390 @@
+
+		if ( self::MIME_TYPE !== $mime_type ) {
+			return $metadata;
+		}
+		$file = get_attached_file( $attachment_id );
+		if ( false === $file ) {
+			return $metadata;
+		}
+
+		$size = $this->get_svg_size( $file );
+		// Check if image size failed to generate and return if so.
+		if ( is_wp_error( $size ) ) {
+			return $metadata;
+		}
+
+		return [
+			'width'    => $size['width'],
+			'height'   => $size['height'],
+			'file'     => _wp_relative_upload_path( $file ),
+			'filesize' => (int) filesize( $file ),
+			'sizes'    => [],
+		];
+	}
+
+	/**
+	 * Hook into upload and error if size could not be generated.
+	 *
+	 * @param array $upload {
+	 *      Array of upload data.
+	 *
+	 *      @type string $file   Filename of the newly-uploaded file.
+	 *      @type string $url    URL of the newly-uploaded file.
+	 *      @type string $type   Mime type of the newly-uploaded file.
+	 * }
+	 *
+	 * @return string[]
+	 */
+	public function wp_handle_upload( $upload ) {
+		if ( self::MIME_TYPE !== $upload['type'] ) {
+			return $upload;
+		}
+
+		$sanitized = $this->sanitize( $upload['tmp_name'] );
+		if ( is_wp_error( $sanitized ) ) {
+			return [ 'error' => $sanitized->get_error_message() ];
+		}
+
+		$size = $this->get_svg_size( $upload['tmp_name'] );
+		if ( is_wp_error( $size ) ) {
+			return [ 'error' => $size->get_error_message() ];
+		}
+
+		return $upload;
+	}
+
+
+	/**
+	 * Get SVG image size.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 *
+	 * @param string $file Path to SVG file.
+	 *
+	 * @return array|WP_Error
+	 */
+	protected function get_svg_size( $file ) {
+		$svg = $this->get_svg_data( $file );
+		$xml = $this->get_xml( $svg );
+
+		if ( false === $xml ) {
+			return new WP_Error( 'invalid_xml_svg', __( 'Invalid xml in SVG.', 'web-stories' ) );
+		}
+
+		$attributes = $xml->attributes();
+		$width      = isset( $attributes->width ) ? (int) $attributes->width : 0;
+		$height     = isset( $attributes->height ) ? (int) $attributes->height : 0;
+
+		// If height and width are not set, try the viewport attribute.
+		if ( ! $width || ! $height ) {
+			$view_box = isset( $attributes->viewBox ) ? (string) $attributes->viewBox : '';
+			if ( empty( $view_box ) ) {
+				$view_box = isset( $attributes->viewbox ) ? (string) $attributes->viewbox : '';
+			}
+			$pieces = explode( ' ', $view_box );
+			if ( count( $pieces ) === 4 ) {
+				$width  = $pieces[2];
+				$height = $pieces[3];
+			}
+		}
+
+		if ( ! $width || ! $height ) {
+			return new WP_Error( 'invalid_svg_size', __( 'Unable to generate SVG image size.', 'web-stories' ) );
+		}
+
+		return compact( 'width', 'height' );
+	}
+
+	/**
+	 * Sanitize the SVG
+	 *
+	 * @param string $file File path.
+	 *
+	 * @return true|WP_Error
+	 */
+	protected function sanitize( $file ) {
+		$dirty     = $this->get_svg_data( $file );
+		$sanitizer = new Sanitizer();
+		$clean     = $sanitizer->sanitize( $dirty );
+
+		if ( false == $clean ) {
```

Comment:
Let's use `===`

Reply:
```suggestion
		if ( false === $clean ) {
```

Reply:
I think this failed for some reason. Let's try again.